opencti-platform/opencti-front/src/private/components/data/playbooks/utils/playbook.ts

```diff
@@ -0,0 +1,179 @@
+import { Edge } from 'reactflow';
+interface PortDefinition {
+  id: string
+  type: string
+}
+interface PlaybookComponent {
+    readonly configuration_schema?: string | null;
+    readonly description: string;
+    readonly icon: string;
+    readonly id: string;
+    readonly is_entry_point?: boolean | null;
+    readonly is_internal?: boolean | null;
+    readonly name: string;
+    readonly ports:  readonly PortDefinition[];
+}
+
+interface LinkDefinition {
+  id: string,
+  from: {
+    port: string,
+    id: string
+  },
+  to: {
+    id: string
+  }
+}
+interface ComputedNodesReturns {
+    id: string;
+    type: string;
+    position: { x: number, y: number };
+    data: {
+        name: string;
+        configuration: string; // json
+        component?: PlaybookComponent | null;
+        openConfig: (nodeId: string) => void;
+        openReplace: (nodeId: string) => void;
+        openAddSibling: (nodeId: string) => void;
+        openDelete: (nodeId: string) => void;
+    };
+}
+interface ComputeNodeDefinition {
+  id: string,
+  name: string,
+  position: { x: number, y: number },
+  component_id: string,
+  configuration: string // json
+}
+
+export const computeNodes = (
+  playbookNodes: ComputeNodeDefinition[],
+  playbookComponents:readonly (PlaybookComponent | null | undefined)[],
```

Comment:
maybe we can avoid the readonly here as well with the right generated type from query